src/main/java/com/gmail/nossr50/skills/woodcutting/WoodcuttingManager.java

```diff
@@ -322,13 +322,17 @@ private void dropTreeFellerLootFromBlocks(@NotNull Set<BlockState> treeFellerBlo
                 processBonusDropCheck(blockState);
             } else if (BlockUtils.isNonWoodPartOfTree(blockState)) {
                 // 75% of the time do not drop leaf blocks
-                if (blockState.getType().getKey().getKey().toLowerCase().contains("sapling")
-                        || ThreadLocalRandom.current().nextInt(100) > 75) {
+                if (ThreadLocalRandom.current().nextInt(100) > 75) {
                     Misc.spawnItemsFromCollection(getPlayer(),
                             Misc.getBlockCenter(blockState),
                             block.getDrops(itemStack),
                             ItemSpawnReason.TREE_FELLER_DISPLACED_BLOCK);
                 }
+                // drop saplings as occur in rest if Knock on Wood unlocked.
+                else if (RankUtils.hasUnlockedSubskill(player, SubSkillType.WOODCUTTING_KNOCK_ON_WOOD)) {
+                    Misc.spawnItemIfSapling(getPlayer(), Misc.getBlockCenter(blockState),
```

Comment:
if ```player``` is already in scope, you don't need ```getPlayer()``` and ```player``` combined, they are the same reference

Reply:
Fixing there and in the existing code above it.

Reply:
fixed